python/dgl/distributed/graph_partition_book.py

```diff
@@ -408,101 +494,238 @@ class RangePartitionBook(GraphPartitionBook):
         partition id of current partition book
     num_parts : int
         number of total partitions
-    node_map : tensor
-        map global node id to partition id
-    edge_map : tensor
-        map global edge id to partition id
+    node_map : a dict of tensors
+        map global node id to partition id for each node type.
+        The key of the dict is the node type and the value is a 2-column matrix.
+        Each row represents the Id range of a partition, the first element is the starting Id
+        and the second element is the ending Id.
+    edge_map : a dict of tensors
+        map global edge id to partition id for each edge type.
+        The key of the dict is the edge type and the value is a 2-column matrix.
+        Each row represents the Id range of a partition, the first element is the starting Id
+        and the second element is the ending Id.
+    ntypes : a dict
+        map ntype strings to ntype Ids.
+    etypes : a dict
+        map etype strings to etype Ids.
     """
-    def __init__(self, part_id, num_parts, node_map, edge_map):
+    def __init__(self, part_id, num_parts, node_map, edge_map, ntypes, etypes):
         assert part_id >= 0, 'part_id cannot be a negative number.'
         assert num_parts > 0, 'num_parts must be greater than zero.'
         self._partid = part_id
         self._num_partitions = num_parts
-        if not isinstance(node_map, np.ndarray):
-            node_map = F.asnumpy(node_map)
-        if not isinstance(edge_map, np.ndarray):
-            edge_map = F.asnumpy(edge_map)
-        self._node_map = node_map
-        self._edge_map = edge_map
+        self._ntypes = [None] * len(ntypes)
+        self._etypes = [None] * len(etypes)
+        for ntype in ntypes:
+            ntype_id = ntypes[ntype]
+            self._ntypes[ntype_id] = ntype
+        assert all([ntype is not None for ntype in self._ntypes]), \
```

Comment:
You could just use `all(self.ntypes)`. Boolean and between None and objects become None and None evaluates to be False.

Reply:
you mean both `None` will be considered as `False`?

Reply:
Yes, that's python standard.